wordpress-mobile/AztecEditor-Android
# Fixing toolbar highlight issues

#395 need's to be merged before this one.

Fixes #345
Fixes #357 
Preserves selected style when deleting the last character on 0 index, and allows the selected style to be removed with extra backspace.

### Test 1

1. Open the demo app.
2. Remove the demo content.
3. Close the device keyboard. (Note that the cursor remains in the text view.)
4. Tap a format toolbar button for inline style (e.g. Bold) to enable it.
5. Tap back into the text view. 
6. Confirm that previously selected style in the toolbar is still highlighted.
7. Type text. Confirm that it's correctly styled.

### Test 2

1. Open the demo app.
2. Remove the demo content.
3. Tap a format toolbar button for inline style(e.g. Bold) to enable it.
4. Type text and press enter.
5. Deselect style and type text.
6. Notice that the style button is not highlighted and style is not applied.

### Test 3

1. Open the demo app.
2. Remove the demo content.
3. Tap a format toolbar button for inline style(e.g. Bold) to enable it.
4. Type text.
5. Press backspace to delete all the text.
6. Confirm that selected style is still highlighted.
7. Press backspace.
8. Confirm that selected style is not highlighted.


There are a couple of changes to inline style handling, so it would be awesome if both @0nko and @theck13 could play with inline styles a bit to make sure everything is ok.
